src/PriceFormatter.php

```diff
@@ -22,4 +24,44 @@ public static function format(float $decimal, Currency $currency, array $overrid
 
         return $currency->prefix() . $decimal . $currency->suffix();
     }
+
+    /** Extract the decimal from the formatted string as per the currency's specifications. */
+    public static function resolve(string $formatted, Currency $currency, array $overrides = []): float
+    {
+        $currency = Currency::fromArray(
+            array_merge(currency($currency)->toArray(), $overrides)
+        );
+
+        $formatted = ltrim($formatted, $currency->prefix());
+        $formatted = rtrim($formatted, $currency->suffix());
+
+        $removeNonDigits = preg_replace('/[^\d' . preg_quote($currency->decimalSeparator()) . ']/', '', $formatted);
+
+        if (! is_string($removeNonDigits)) {
+            throw new \Exception('The formatted string could not be resolved to a valid number.');
+        }
+
+        return (float) str_replace($currency->decimalSeparator(), '.', $removeNonDigits);
+    }
+
+    /** Tries to extract the currency from the formatted string. */
+    public static function extractCurrency(string $formatted): Currency
+    {
+        $possibleCurrency = null;
+
+        foreach (currencies()->all() as $currency) {
+            if (
+                str_starts_with($formatted, $currency->prefix())
+                && str_ends_with($formatted, $currency->suffix())
+            ) {
+                if ($possibleCurrency) {
+                    throw new CannotExtractCurrencyException('Multiple currencies are using the same prefix and suffix. Please specify the currency of the formatted string.');
+                }
+
+                $possibleCurrency = $currency;
+            }
+        }
+
+        return $possibleCurrency ?? throw new CannotExtractCurrencyException('None of the currencies are using the prefix and suffix that would match with the formatted string.');
```

Comment:
1. Move the string to `CannotExtractCurrencyException` constructor
2. Pass the `$formatted` as an argument and make it part of the exception message

Reply:
`CannotExtractCurrencyException` is used twice with a different message. Do you suggest making separate exception classes for them?

![image](https://user-images.githubusercontent.com/11887260/160973649-3378782e-fbf1-469f-905c-04ef23cb6e5d.png)

Reply:
Ah I see. We can use one class then, but show the used value in the message.

Reply:
Done